src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,152 @@
+            }
+            current.next = new Node<>(key, value);
+        }
+
+        size++;
+        sizeCheck();
     }
 
     @Override
     public V getValue(K key) {
+        if (key == null) {
+            return handleNullKey();
+        }
+
+        int bucketIndex = getBucketIndex(key);
+        Node<K, V> current = bucketArray[bucketIndex];
+        while (current != null) {
+            if (key.equals(current.key)) {
+                return current.value;
+            }
+            current = current.next;
+        }
+
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void handleNullKey(V value) {
+        if (bucketArray[0] == null) {
+            bucketArray[0] = new Node<>(null, value); // Add a new node with null key
+            size++;
+            if ((double) size / bucketArray.length > LOAD_FACTOR) {
+                resize();
+            }
+            return;
+        }
+
+        Node<K, V> current = bucketArray[0];
+        while (true) {
+            if (current.key == null) {
+                current.value = value;
+                return;
+            }
+            if (current.next == null) {
+                break;
+            }
+            current = current.next;
+        }
+
+        current.next = new Node<>(null, value);
+        size++;
+    }
+
+    private V handleNullKey() {
+        Node<K, V> current = bucketArray[0];
+        while (current != null) {
+            if (current.key == null) {
+                return current.value;
+            }
+            current = current.next;
+        }
+        return null;
+    }
+
+    private int getBucketIndex(K key) {
+        return (key == null) ? 0 : Math.abs(key.hashCode() % bucketArray.length);
+    }
+
+    private void sizeCheck() {
+        if ((double) size / bucketArray.length > LOAD_FACTOR) {
+            resize();
+        }
+    }
+
+    private void resize() {
+        int newCapacity = bucketArray.length * 2;
+        Node<K, V>[] newBucketArray = new Node[newCapacity];
+
+        for (Node<K, V> kvNode : bucketArray) {
+            Node<K, V> current = kvNode;
+            while (current != null) {
+                int newBucketIndex = (current.key == null)
+                        ? 0 : Math.abs(current.key.hashCode() % newCapacity);
+                if (newBucketArray[newBucketIndex] == null) {
+                    newBucketArray[newBucketIndex] = new Node<>(current.key, current.value);
+                } else {
+                    Node<K, V> newCurrent = newBucketArray[newBucketIndex];
+                    while (newCurrent.next != null) {
+                        newCurrent = newCurrent.next;
+                    }
+                    newCurrent.next = new Node<>(current.key, current.value);
+                }
+                current = current.next;
+            }
+        }
+
+        bucketArray = newBucketArray;
```

Comment:
Method is overcomplicated, should be something like below
```suggestion
        Node<K, V>[] oldBucketArray = bucketArray;
        bucketArray = new Node[oldBucketArray.length * 2];
        size = 0;

        for (Node<K, V> kvNode : oldBucketArray) {
            while (kvNode != null) {
                put(kvNode.key, kvNode.value);
                kvNode = kvNode.next;
            }
        }
```